common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -85,7 +85,7 @@ public void recycle(Object object) {
         // By default we allow one push to a Recycler for each 8th try on handles that were never recycled before.
         // This should help to slowly increase the capacity of the recycler while not be too sensitive to allocation
         // bursts.
-        RATIO = safeFindNextPositivePowerOfTwo(SystemPropertyUtil.getInt("io.netty.recycler.ratio", 8));
+        RATIO = max(0, SystemPropertyUtil.getInt("io.netty.recycler.ratio", 8));
```

Comment:
as the ratio is used with bitmask operations we need to have it either be 0 or have it be a `power of two`. So this change and the one below is not correct

Reply:
This commit 17bffce90e2051250be7eea85a3e327717a6aa7f remove the use of ratioMask. So it doesn't need to `power of two` now?

Reply:
ah sorry... I looked at the old code. You are right.